include/boost/math/fft/dft_api.hpp

```diff
@@ -398,11 +445,16 @@
       OutputIteratorType out)
     {
       resize(std::distance(in_first,in_last));
-      executor_r2c::execute(in_first,in_last,out,
-        [this](const real_type* i, complex_type* o)
+      
+      std::vector<real_type,allocator_t> hc(size(),alloc);
+      
+      executor_halfcomplex::execute(in_first,in_last,hc.data(),
+        [this](const real_type* i, real_type* o)
         {
-          backend::template real_to_complex<complex_type>(i,o);
+          backend::real_to_halfcomplex(i,o);
```

Comment:
On the real transforms, is natural to work from R^N space to 1/2 C^N space and back. Any more degrees of freedom on the complex side would be spurious. Hence my rfft api is built based on R^N -> 1/2 C^N. The complex type in the output is deduced from the iterator traits. This is related to issue #14.